aiida/manage/configuration/settings.py

```diff
@@ -87,7 +88,9 @@ def set_configuration_directory():
 
     environment_variable = os.environ.get(DEFAULT_AIIDA_PATH_VARIABLE, None)
 
-    if environment_variable:
+    if aiida_config_folder is not None:
+        AIIDA_CONFIG_FOLDER = aiida_config_folder
```

Comment:
Potentially also do this to make behaviour similar to env var
```suggestion
        AIIDA_CONFIG_FOLDER = aiida_config_folder.expanduser()
```

Reply:
Since this is only accessible from the API where a caller can call this if needed (and should be only internal anyway) think it is fine to leave as is.